### PR TITLE
fix error log

### LINKED ISSUE #650: Too many error messages of Kmesh daemon log
**What would you like to be added**:

Don't output unnecessary error logs

**Why is this needed**:

When default ns is managed by Kmesh, If I delete all pods under that ns, the following log will be reported:

```bash
time="2024-07-30T02:32:27Z" level=info msg="default/sleep-5577c64d7c-f4jwg: Pod managed by Kmesh is deleted" subsys=manage_controller
time="2024-07-30T02:32:27Z" level=info msg="default/sleep-5577c64d7c-qxwgs: enable Kmesh manage" subsys=manage_controller
time="2024-07-30T02:32:27Z" level=error msg="failed to enable Kmesh manage" subsys=manage_controller
time="2024-07-30T02:32:27Z" level=info msg="default/httpbin-55ffdcd94-c5v2l: enable Kmesh manage" subsys=manage_controller
time="2024-07-30T02:32:27Z" level=error msg="failed to enable Kmesh manage" subsys=manage_controller
time="2024-07-30T02:32:27Z" level=info msg="default/sleep-5577c64d7c-lc4d9: Pod managed by Kmesh is deleted" subsys=manage_controller
time="2024-07-30T02:32:27Z" level=info msg="default/sleep-5577c64d7c-qxwgs: enable Kmesh manage" subsys=manage_controller
time="2024-07-30T02:32:27Z" level=error msg="failed to enable Kmesh manage" subsys=manage_controller
time="2024-07-30T02:32:27Z" level=info msg="default/httpbin-55ffdcd94-c5v2l: enable Kmesh manage" subsys=manage_controller
time="2024-07-30T02:32:27Z" level=error msg="failed to enable Kmesh manage" subsys=manage_controller
time="2024-07-30T02:32:27Z" level=info msg="default/sleep-5577c64d7c-dg5rk: enable Kmesh manage" subsys=manage_controller
time="2024-07-30T02:32:27Z" level=error msg="failed to enable Kmesh manage" subsys=manage_controller
time="2024-07-30T02:32:27Z" level=info msg="default/sleep-5577c64d7c-dg5rk: enable Kmesh manage" subsys=manage_controller
time="2024-07-30T02:32:27Z" level=error msg="failed to enable Kmesh manage" subsys=manage_controller
time="2024-07-30T02:32:28Z" level=info msg="default/httpbin-55ffdcd94-c5v2l: enable Kmesh manage" subsys=manage_controller
time="2024-07-30T02:32:28Z" level=error msg="failed to enable Kmesh manage" subsys=manage_controller
time="2024-07-30T02:32:28Z" level=info msg="default/sleep-5577c64d7c-qxwgs: enable Kmesh manage" subsys=manage_controller
time="2024-07-30T02:32:28Z" level=error msg="failed to enable Kmesh manage" subsys=manage_controller
time="2024-07-30T02:32:28Z" level=info msg="default/sleep-5577c64d7c-dg5rk: enable Kmesh manage" subsys=manage_controller
time="2024-07-30T02:32:28Z" level=error msg="failed to enable Kmesh manage" subsys=manage_controller
time="2024-07-30T02:32:28Z" level=info msg="default/httpbin-55ffdcd94-7kz6x: Pod managed by Kmesh is deleted" subsys=manage_controller
```

It seems that there is nothing abnormal, but there are still many errro logs in Kmesh daemon, which will make users very confused.

Then I let the applications access each other and there will be the following error log:

```bash
time="2024-07-30T02:44:37Z" level=error msg="get bpf map of metric FAILED, err:can't marshal key: *telemetry.metricKey doesn't marshal to 40 bytes" subsys=pkg/telemetry
time="2024-07-30T02:44:37Z" level=error msg="get bpf map of metric FAILED, err:can't marshal key: *telemetry.metricKey doesn't marshal to 40 bytes" subsys=pkg/telemetry
time="2024-07-30T02:44:37Z" level=error msg="get bpf map of metric FAILED, err:can't marshal key: *telemetry.metricKey doesn't marshal to 40 bytes" subsys=pkg/telemetry
time="2024-07-30T02:44:37Z" level=error msg="get bpf map of metric FAILED, err:can't marshal key: *telemetry.metricKey doesn't marshal to 40 bytes" subsys=pkg/telemetry
```

Still very confusing and don't know how to deal with it.

### PR DESCRIPTION
**Pull request by weli-l**
**What type of PR is this?**

**What this PR does / why we need it**:
fix error log
**Which issue(s) this PR fixes**:
Fixes #650

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by hzxuzhonghu — COMMENTED** [review]
I would prefer moving the check into `HandleKmeshManage`, and check pod status there and also get nsPath there

**Comment by weli-l** [comment]
> I would prefer moving the check into `HandleKmeshManage`, and check pod status there and also get nsPath there

cni also use handleKmeshManage， if move getnspath into handleKmeshManage, logic seems a bit confusing

**Comment by hzxuzhonghu** [comment]
Ah, i forget that

File: pkg/controller/manage/manage_controller.go [hunk 1]
```diff
@@ -113,7 +128,10 @@ func NewKmeshManageController(client kubernetes.Interface, security *kmeshsecuri
 				log.Errorf("expected *corev1.Pod but got %T and %T", oldObj, newObj)
 				return
 			}
-
+			if !isPodReady(newPod) {
+				log.Debugf("%s/%s: Pod is not ready, skipping Kmesh manage enable", newPod.GetNamespace(), newPod.GetName())
```
Comment (YaoZengzeng) [inline pkg/controller/manage/manage_controller.go:132]: Maybe we could add prefix to the log to distinguish the processing of different pod event, like "pod add event: %s/%s...."

File: pkg/controller/manage/manage_controller.go [hunk 2]
```diff
@@ -86,6 +95,12 @@ func NewKmeshManageController(client kubernetes.Interface, security *kmeshsecuri
 				log.Errorf("expected *corev1.Pod but got %T", obj)
 				return
 			}
+
+			if !isPodReady(pod) {
+				log.Debugf("%s/%s: Pod is not ready, skipping Kmesh manage enable", pod.GetNamespace(), pod.GetName())
```
Comment (YaoZengzeng) [inline pkg/controller/manage/manage_controller.go:100]: Add log prefix as well.

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm

**Pull request merged**